### ISSUE TITLE
Null ref in reading ApiDescriptions with null HttpMethod

### ISSUE THREAD

**Issue by NTaylorMullen** [domaindrivendev/Ahoy#6]
**Steps:**

1. File New Web Application => Web Site
1. Add reference to swaggerui in project.json
1. Add swagger/swaggerui to startup.cs
1. Add the controller pasted below.
1. hit the /swagger/v1/swagger.json endpoint and see the error pasted below.

**Controller:**

```C#
[Route("Something")]
[Produces("application/json")]
public class AnnotatedController : Controller
{
    [Route("SomethingIndex")]
    public IActionResult Index()
    {
        return View();
    }
}
```

**Call stack with extra info in it:**
```
Swashbuckle.Swagger.SwaggerGenerator.<>c.<CreatePathItem>b__6_0(ApiDescription apiDesc) in SwaggerGenerator.cs
        private PathItem CreatePathItem(IEnumerable<ApiDescription> apiDescriptions, ISchemaRegistry schemaRegistry)
        {
            var pathItem = new PathItem();
            // Group further by http method
            var perMethodGrouping = apiDescriptions
                .GroupBy(apiDesc => apiDesc.HttpMethod.ToLower());
            foreach (var group in perMethodGrouping)
            {
                var httpMethod = group.Key;
                if (group.Count() > 1) throw new NotSupportedException(string.Format(
System.Linq.Lookup`2.Create[TSource](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
System.Linq.GroupedEnumerable`3.GetEnumerator()
Swashbuckle.Swagger.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, ISchemaRegistry schemaRegistry) in SwaggerGenerator.cs
            var pathItem = new PathItem();
            // Group further by http method
            var perMethodGrouping = apiDescriptions
                .GroupBy(apiDesc => apiDesc.HttpMethod.ToLower());
            foreach (var group in perMethodGrouping)
            {
                var httpMethod = group.Key;
                if (group.Count() > 1) throw new NotSupportedException(string.Format(
                    "Not supported by Swagger 2.0: Multiple operations with path '{0}' and method '{1}'.",
                    group.First().RelativePathSansQueryString(), httpMethod));
Swashbuckle.Swagger.SwaggerGenerator.<>c__DisplayClass4_0.<GetSwagger>b__4(IGrouping`2 group) in SwaggerGenerator.cs
                throw new UnknownApiVersion(apiVersion);
            var paths = GetApiDescriptionsFor(apiVersion)
                .Where(apiDesc => !(_options.IgnoreObsoleteActions && apiDesc.IsObsolete()))
                .OrderBy(_options.GroupNameSelector, _options.GroupNameComparer)
                .GroupBy(apiDesc => apiDesc.RelativePathSansQueryString())
                .ToDictionary(group => "/" + group.Key, group => CreatePathItem(group, schemaRegistry));
            var swaggerDoc = new SwaggerDocument
            {
                Info = info,
                Host = _options.Host ?? defaultHost,
                BasePath = _options.BasePath ?? defaultBasePath,
System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
Swashbuckle.Swagger.SwaggerGenerator.GetSwagger(String apiVersion, String defaultHost, String defaultBasePath, String[] defaultSchemes) in SwaggerGenerator.cs
            var schemaRegistry = _schemaRegistryFactory();
            var info = _options.ApiVersions.FirstOrDefault(v => v.Version == apiVersion);
            if (info == null)
                throw new UnknownApiVersion(apiVersion);
            var paths = GetApiDescriptionsFor(apiVersion)
                .Where(apiDesc => !(_options.IgnoreObsoleteActions && apiDesc.IsObsolete()))
                .OrderBy(_options.GroupNameSelector, _options.GroupNameComparer)
                .GroupBy(apiDesc => apiDesc.RelativePathSansQueryString())
                .ToDictionary(group => "/" + group.Key, group => CreatePathItem(group, schemaRegistry));
            var swaggerDoc = new SwaggerDocument
Swashbuckle.Application.SwaggerDocsMiddleware.<Invoke>d__5.MoveNext() in SwaggerDocsMIddleware.cs
            if (!RequestingSwaggerDocs(httpContext.Request, out apiVersion))
            {
                await _next(httpContext);
                return;
            }
            var swagger = _swaggerProvider.GetSwagger(apiVersion, null, httpContext.Request.PathBase);
            RespondWithSwaggerJson(httpContext.Response, swagger);
        }
        private bool RequestingSwaggerDocs(HttpRequest request, out string apiVersion)
        {
            apiVersion = null;
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.AspNet.Authentication.AuthenticationMiddleware`1.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Diagnostics.Entity.DatabaseErrorPageMiddleware.<Invoke>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.AspNet.Diagnostics.Entity.DatabaseErrorPageMiddleware.<Invoke>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
Microsoft.AspNet.Diagnostics.ErrorPageMiddleware.<Invoke>d__4.MoveNext()
```

**Comment by MiguelAlho** [comment]
+1 getting this too

**Comment by domaindrivendev** [comment]
If you don't decorate an action with a specific Http verb, then WebApi will make that action available through ALL of the Http verbs. Swashbuckle aside, is this what you want to do from an API perspective?

If so, I'd be interested to hear the rationale. I'd also be interested to how you think Swashbuckle should behave in this case. Some options are ...

1) Ignore the operation
2) Default to GET
3) Create a description of the action for each Http verb
4) Throw a more informative exception

**Comment by NTaylorMullen** [comment]
@domaindrivendev First question: I'd think from an API perspective it's definitely odd, but the user may have opted into it. My personal opinion is to do a smart variation of 3; smart in the sense of, generate descriptors that don't conflict with other APIs on the controller. Ultimately this would reflect what a user could consume and how they'd do it.

Swashbuckle is a fantastic tool for inspecting what the API is capable of. Heck, if anything you may help  users catch some errors by exposing exactly how the API can be accessed.

**Comment by zoranpro** [comment]
I started to get a similar error as well. Not sure why that started to happen. Here is the error: https://gist.github.com/zoranpro/b05fe89569992f0d1183

**Comment by rytmis** [comment]
At the very least, 4) would help me notice that I've neglected to define the verb. :)

**Comment by lfoust** [comment]
I am getting this error from a non-api endpoint. My Asp.Net 5 project is a mix of API endpoints with some regular MVC controller/action/views used for admin purposes.

I think my vote in this situation would be to default to GET and to somehow add a warning so developers can decide if they are happy with the default behavior or if they want to be explicit about the Http Method.

**Comment by domaindrivendev** [comment]
Went with a more informative exception for the time now. Available in 6.0.0-rc1-final
